### PR TITLE
chunkserver: fix crc issue

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/chunkserver/scan_manager.cpp [hunk 1]
```diff
@@ -300,14 +302,19 @@ void ScanManager::SetLocalScanMap(ScanKey key, ScanMap map) {
                      << " copysetId = " << key.second;
         return;
     }
-    if (job->currentChunkId != map.chunkid() ||
-        job->currentOffset != map.offset()) {
+    job->taskLock.RDLock();
```
Comment (Wine93) [inline src/chunkserver/scan_manager.cpp:305]: Maybe the logic of lock can be simplify: 
```
job->taskLock.RDLock();
bool matched = job->task.chunkId != map.chunkid() ||
               job->task.offset != map.offset() || 
               job->task.len != map.len();
job->taskLock.Unlock();

if (matched) {
    ...
} else {
    ...
}
```

File: src/chunkserver/scan_manager.cpp [hunk 2]
```diff
@@ -321,8 +328,19 @@ void ScanManager::DealFollowerScanMap(const FollowScanMapRequest &request,
...
+    if (nullptr == job) {
+        LOG(WARNING) << "DealFollowerScanMap failed, job not found."
+                     << " logical poolId = " << key.first
+                     << " copysetId = " << key.second;
+        response->set_retcode(CHUNK_OP_STATUS::CHUNK_OP_STATUS_INVALID_REQUEST);
+        return;
+    }
+
+    job->taskLock.RDLock();
+    if (job->task.chunkId == scanMap.chunkid() &&
```
Comment (Wine93) [inline src/chunkserver/scan_manager.cpp:340]: ditto

File: src/chunkserver/scan_manager.cpp (hunk 1 above)
Comment (SeanHai) [reply src/chunkserver/scan_manager.cpp:305]: > Maybe the logic of lock can be simplify:
> 
> ```
> job->taskLock.RDLock();
> bool matched = job->task.chunkId != map.chunkid() ||
>                job->task.offset != map.offset() || 
>                job->task.len != map.len();
> job->taskLock.Unlock();
> 
> if (matched) {
>     ...
> } else {
>     ...
> }
> ```

done

File: src/chunkserver/scan_manager.cpp (hunk 2 above)
Comment (SeanHai) [reply src/chunkserver/scan_manager.cpp:340]: > ditto

done

**Review by Wine93 — APPROVED** [review]

**Review by wu-hanqing — APPROVED** [review]

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**